backend/wordAlgorithm/word.py

```diff
@@ -0,0 +1,94 @@
+# Format of xml taken as input:
+#    <Words>
+#        <Word>
+#            <Language>Dutch</Language>
+#            <Word>ja</Word>
+#            <Translation>yes</Translation>
+#            <Category/>
+#            <Word_Difficulty/>
+#            <Extensions/>
+#        </Word>
+#    </Words>
+
+
+words = []
+
+# Word object
+class Word(object):
+    originalWord = "";
+    language = ""
+    translatedWord = "";
+    category = "";
+    wordValue = 0;
+    extensions = []
+    progress = 0;
+    wordDifficulty = 0;
+    
+# Word constructor
+def makeWord(language, originalWord, translatedWord, category, extensions, wordDifficulty):
+    word = Word()
+    word.originalWord = originalWord
+    word.translatedWord = translatedWord
+    word.category = category
+    word.extensions = extensions
+    word.wordDifficulty = wordDifficulty
+    word.language = language
+    return word
+
+# Returns the value if
+def checkExistance (value):
+    if not (value is None):
+        return value
+    return ""
+
+# Word printer
+def printWord(word):
+    print 
+    print "Language: ", word.language
+    print "Original Word: ", word.originalWord
+    print "Translated Word: ", word.translatedWord
+    print "Category: ", word.category
+    print "Extensions: ", word.extensions
+    print "Word Difficulty: ", word.wordDifficulty
+
+
+language = ""
+originalWord = ""
+translatedWord = ""
+category = 0
+extensions = ""
+wordDifficulty = ""
+
+# XML parsing
+import xml.etree.ElementTree as ET
+tree = ET.parse('words.xml');
+root = tree.getroot()
+for child in root:
+    for child1 in child:
```

Comment:
Imports should be at the top. Line 63 to 65 seem copied/pasted if that is the case reference the source in the comments